presto-raptorx/src/main/java/com/facebook/presto/raptorx/RaptorMetadata.java

```diff
@@ -0,0 +1,742 @@
+
+        table.getTemporalColumn().ifPresent(column -> properties.put(TEMPORAL_COLUMN_PROPERTY, column.getColumnName()));
+
+        if (!bucketing.isEmpty()) {
+            properties.put(BUCKETED_ON_PROPERTY, bucketing);
+        }
+        if (!ordering.isEmpty()) {
+            properties.put(ORDERING_PROPERTY, ordering);
+        }
+
+        properties.put(BUCKET_COUNT_PROPERTY, distribution.getBucketCount());
+        distribution.getDistributionName().ifPresent(name -> properties.put(DISTRIBUTION_NAME_PROPERTY, name));
+
+        List<ColumnMetadata> columns = table.getColumns().stream()
+                .map(RaptorMetadata::toColumnMetadata)
+                .collect(toCollection(ArrayList::new));
+
+        columns.add(toColumnMetadata(CHUNK_ID_HANDLE));
+
+        if (!bucketing.isEmpty()) {
+            columns.add(toColumnMetadata(BUCKET_NUMBER_HANDLE));
+        }
+
+        return new ConnectorTableMetadata(
+                new SchemaTableName(schema.getSchemaName(), table.getTableName()),
+                columns,
+                properties.build(),
+                table.getComment());
+    }
+
+    @Override
+    public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> schemaName)
+    {
+        return schemaName.map(transaction::listTables)
+                .orElseGet(transaction::listTables);
+    }
+
+    @Override
+    public Map<String, ColumnHandle> getColumnHandles(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        TableInfo table = transaction.getTableInfo(((RaptorTableHandle) tableHandle).getTableId());
+
+        List<RaptorColumnHandle> columns = table.getColumns().stream()
+                .map(RaptorMetadata::toColumnHandle)
+                .collect(toCollection(ArrayList::new));
+
+        columns.add(CHUNK_ID_HANDLE);
+
+        if (!table.getBucketColumns().isEmpty()) {
+            columns.add(BUCKET_NUMBER_HANDLE);
+        }
+
+        return columns.stream().collect(toImmutableMap(RaptorColumnHandle::getColumnName, identity()));
+    }
+
+    @Override
+    public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle columnHandle)
+    {
+        return toColumnMetadata((RaptorColumnHandle) columnHandle);
+    }
+
+    @Override
+    public Map<SchemaTableName, List<ColumnMetadata>> listTableColumns(ConnectorSession session, SchemaTablePrefix prefix)
+    {
+        List<SchemaTableName> tableNames;
+        if (prefix.getTableName() != null) {
+            tableNames = ImmutableList.of(new SchemaTableName(prefix.getSchemaName(), prefix.getTableName()));
+        }
+        else {
+            tableNames = listTables(session, Optional.ofNullable(prefix.getSchemaName()));
+        }
+
+        // TODO: this should fetch all tables at once
+        ImmutableMap.Builder<SchemaTableName, List<ColumnMetadata>> map = ImmutableMap.builder();
```

Comment:
"highker on Feb 4  Member
Ya, just add a new method to dao; same for views"

As the 'TODO' said, it should have a "dao" that fetch all tables at once. For now I will keep the 'TODO', as mentioned above, a general task  "complete RepeatableRead" would cover it; maybe no need for "complete RR" after all, wait to see